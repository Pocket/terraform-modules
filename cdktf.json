{
  "language": "typescript",
  "app": "npm run --silent compile && node dist/src/example.js",
  "terraformProviders": [
    "pagerduty/pagerduty@~> 1.8.0",
    "archive@~> 1.3.0",
    "local@~> 2.1.0"
  ]
}
