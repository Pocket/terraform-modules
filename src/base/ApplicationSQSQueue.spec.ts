import { Testing, TerraformStack } from 'cdktf';
import {
  ApplicationSQSQueue,
  ApplicationSQSQueueProps,
} from './ApplicationSQSQueue';

describe('ApplicationSQSQueue', () => {
  it('renders an sqs queue without tags', () => {
    const synthed = Testing.synthScope((stack) => {
      new ApplicationSQSQueue(stack, 'testSQS', {
        name: 'TEST',
      });
    });
    expect(synthed).toMatchSnapshot();
  });

  it('renders an sqs queue with tags', () => {
    const synthed = Testing.synthScope((stack) => {
      new ApplicationSQSQueue(stack, 'testSQS', {
        name: 'TEST',
        tags: {
          test: '123',
          environment: 'prod',
        },
      });
    });
    expect(synthed).toMatchSnapshot();
  });

  it('renders an sqs queue with max message size', () => {
    const synthed = Testing.synthScope((stack) => {
      new ApplicationSQSQueue(stack, 'testSQS', {
        name: 'TEST',
        maxMessageSize: 86753,
      });
    });
    expect(synthed).toMatchSnapshot();
  });

  const testFieldValidation = (config: ApplicationSQSQueueProps) => {
    const sqsConfig = {
      name: 'test',
      ...config,
    };

    expect(
      () =>
        new ApplicationSQSQueue(
          new TerraformStack(Testing.app(), 'test'),
          'test-sqs',
          sqsConfig
        )
    ).toThrow(Error);
  };

  it('test sqs validators', () => {
    testFieldValidation({ visibilityTimeoutSeconds: -1 });
    testFieldValidation({ visibilityTimeoutSeconds: 43201 });

    testFieldValidation({ messageRetentionSeconds: 59 });
    testFieldValidation({ messageRetentionSeconds: 1209601 });

    testFieldValidation({ maxMessageSize: 234 });
    testFieldValidation({ maxMessageSize: 262145 });

    testFieldValidation({ delaySeconds: -1 });
    testFieldValidation({ delaySeconds: 901 });

    testFieldValidation({ receiveWaitTimeSeconds: -1 });
    testFieldValidation({ receiveWaitTimeSeconds: 24 });
  });
});
