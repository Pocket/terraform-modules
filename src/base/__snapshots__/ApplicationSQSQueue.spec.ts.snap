// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders an sqs queue with max message size 1`] = `
"{
  \\"//\\": {
    \\"metadata\\": {
      \\"version\\": \\"stubbed\\",
      \\"stackName\\": \\"test\\"
    }
  },
  \\"resource\\": {
    \\"aws_sqs_queue\\": {
      \\"testSQS_sqs_queue_B45AAD64\\": {
        \\"fifo_queue\\": false,
        \\"max_message_size\\": 86753,
        \\"name\\": \\"TEST\\",
        \\"//\\": {
          \\"metadata\\": {
            \\"path\\": \\"test/testSQS/sqs_queue\\",
            \\"uniqueId\\": \\"testSQS_sqs_queue_B45AAD64\\"
          }
        }
      }
    }
  }
}"
`;

exports[`renders an sqs queue with tags 1`] = `
"{
  \\"//\\": {
    \\"metadata\\": {
      \\"version\\": \\"stubbed\\",
      \\"stackName\\": \\"test\\"
    }
  },
  \\"resource\\": {
    \\"aws_sqs_queue\\": {
      \\"testSQS_sqs_queue_B45AAD64\\": {
        \\"fifo_queue\\": false,
        \\"name\\": \\"TEST\\",
        \\"tags\\": {
          \\"test\\": \\"123\\",
          \\"environment\\": \\"prod\\"
        },
        \\"//\\": {
          \\"metadata\\": {
            \\"path\\": \\"test/testSQS/sqs_queue\\",
            \\"uniqueId\\": \\"testSQS_sqs_queue_B45AAD64\\"
          }
        }
      }
    }
  }
}"
`;

exports[`renders an sqs queue without tags 1`] = `
"{
  \\"//\\": {
    \\"metadata\\": {
      \\"version\\": \\"stubbed\\",
      \\"stackName\\": \\"test\\"
    }
  },
  \\"resource\\": {
    \\"aws_sqs_queue\\": {
      \\"testSQS_sqs_queue_B45AAD64\\": {
        \\"fifo_queue\\": false,
        \\"name\\": \\"TEST\\",
        \\"//\\": {
          \\"metadata\\": {
            \\"path\\": \\"test/testSQS/sqs_queue\\",
            \\"uniqueId\\": \\"testSQS_sqs_queue_B45AAD64\\"
          }
        }
      }
    }
  }
}"
`;
